swan-lake/learn-the-platform/source-code-dependencies/manage-dependencies.md

```diff
@@ -153,6 +157,65 @@ Ballerina considers the version specified in the Ballerina.toml as the minimum r
 However, the compiler gives priority to the latest version if a new patch version is found in distribution or Ballerina Central repositories.
 At this point, the compiler resolves the latest version and ignores the dependency version in the local repository.
 
+
+## Use dependencies from a custom repository
+
+Ballerina supports [Nexus](https://www.sonatype.com/products/sonatype-nexus-repository), [Artifactory](https://jfrog.com/artifactory/) and [Github packages](https://github.com/features/packages) to be set up as custom repositories. You can configure the custom repository in `<USER_HOME>/.ballerina/Settings.toml` to integrate it to package resolution.
+
+```toml
+[[repository.TYPE]]
+id = "<repository-id>" # This id is used when pushing/ pulling balas
+url = "<repository-url>"
+userId = "<username/ specific-userId>"
+accesstoken = "<password/ accesstoken>"
+```
+Here `TYPE` can be anything among `nexus`,  `artifactory`,  `github`. The following steps show how to configure a GitHub package repository to resolve a specific dependency.
+
+```toml
+[[repository.github]]
+id = "github_1" # This id is used when pushing/ pulling balas
+url = "https://maven.pkg.github.com/ballerina-platform/ballerina-release"
+userId = "jackson12"
+accesstoken = "ghp_nMlJsjshhdtdt5367389920020hHfrdrd"
+```
+
+### Publish the Ballerina library package to a custom repository
+
+1. Generate the Ballerina archive. 
+   ```
+   $ bal pack
+   ```
+
+2. Publish to the custom repository.
+   ```
+   $ bal push --repository <repo-id>
+   ```
+
+   If you already have the path of Ballerina archive, then,
+
+    ```
+    $ bal push --repository <repo-id> <path-to-bala-archive>
+    ```
+
+### Pull the Ballerina library package from a custom repository
+
+1. Ballerina package from the custom remote repository can be pulled using the following command.  
+    ```
+   $ bal pull <org>/<package-name>:<version> --repository <repo-id>
```

Comment:
```suggestion
   $ bal pull <package-org>/<package-name>:<version> --repository <repository-id>
```

The `package-org` is equivalent to the `group-id` whereas the `package-name` is equivalent to the `artifact-id`.